cmd/xl-storage.go

```diff
@@ -2236,6 +2233,11 @@ func (s *xlStorage) RenameData(ctx context.Context, srcVolume, srcPath, dataDir,
 		}
 	}
 
+	// Commit meta-file
+	if err = renameAll(srcFilePath, dstFilePath); err != nil {
+		return osErrToFileErr(err)
```

Comment:
if there is an error here can we purge the `data`?  @vadmeste

Reply:
The reason for the previous pattern is to ensure such that we always have `xl.meta` present to display correctly an object when listing. I am okay with this change but you should purge the data here if `xl.meta` failed.

Reply:
> I am okay with this change but you should purge the data here if `xl.meta` failed.

I don't think purging data is an good idea, since there will be no revert back when it fails.

For example, renameData() function calls RenameData() but undo the operation when no enough disks were able to do this.

Reply:
If you cannot write xl.meta there is no point of having the data which you just created. 

Note: In case of versioning this gets very tricky.

The approach you are taking here prefers data which is not useful because we need to keep xl.meta always the preferred entity for many multiple reasons.

The problem is RenameData is many calls combined and it is not atomic in nature because each of these operations are independent and have independent failures.

xl.meta is the source of truth and source of truth should be persisted first.

Reply:
Actually, after looking more closely, It is okay to take this change as quorum would be present at the xl.meta layer but we also might have parts available to read from.